odl/tomo/data/clintec_spect.py

```diff
@@ -0,0 +1,152 @@
+
+
+def read_clintec_CT_reconstruction(path):
+    """Read DICOM format data to array.
+
+    Parameters
+    ----------
+    path : `str`
+        Path of the folder where DICOM data is stored.
+
+    Returns
+    -------
+    ct_volume : `array-like`
+        Array of CT reconstruction in Houndsfield units.
+
+    dataset : DICOM File Meta Information
+
+    """
+
+    dcm_file_list = []
+    for dir_name, _, file_list in os.walk(path):
+        for filename in sorted(file_list, key=natural_sort_key):
+            dcm_file_list.append(os.path.join(dir_name, filename))
+
+    dataset = dicom.read_file(dcm_file_list[0])
+
+    shape = (int(dataset.Rows), int(dataset.Columns), len(dcm_file_list))
+    data_array = np.empty(shape, dtype=dataset.pixel_array.dtype)
+
+    for i, fname in enumerate(dcm_file_list):
+        ds = dicom.read_file(fname)
+        data_array[:, :, i] = ds.pixel_array
+
+    rescale_intercept = float(dataset.RescaleIntercept)
+    rescale_slope = float(dataset.RescaleSlope)
+
+    ct_volume = rescale_slope * data_array + rescale_intercept
+
+    return ct_volume, dataset
+
+
+def linear_attenuation_from_HU(volume, a, b):
+    """Convert Houndsfield units to linear attenuation coefficient.
+
+    Parameters
+    ----------
+    volume : `array-like`
+        array of Houndsfield units to be converted
+
+    a, b : `sequence`
+        a = (a1, a2) where a1 `float` and a2 `float` are the intercepts
+            of the bilinear curve
+        b = (b1, b2) where b1 `float` and b2 `float` are the slopes
+            of the bilinear curve
+        see Notes_ for furher information.
+
+    Returns
+    -------
+    attenuation : `array-like`
+        array of linear attenuation values.
+
+    Notes
+    -----
+        `a` and `b` are parameters of a bilinear curve for converting
+        Houndsfield units (HU) to linear attenuation coefficients, such that
+
+        mu = a1 + b1 * ct_volume for HU < 0
+
+        mu = a2 + b2 * ct_volume for HU >= 0
+
+        See [Bro+2008]_ and [Bur+2002]_ for example.
+
+    """
+    a1, a2 = a
+    b1, b2 = b
+    mask = np.less(volume, 0)
+    attenuation = np.empty_like(volume)
+    attenuation[mask] = a1 + b1 * volume[mask]
+    attenuation[~mask] = a2 + b2 * volume[~mask]
+    attenuation = np.absolute(attenuation)
+    return attenuation
+
+
+def read_clintec_raw_spect_data(path):
+    """Read raw SPECT DICOM data to array.
+
+    Parameters
+    ----------
+    path : `str`
+        Path of the folder where DICOM data is stored.
+
+    Returns
+    -------
+    spect_data : `array-like`
+        SPECT projection data
+    dataset : DICOM File Meta Information
+    """
+
+    filenames = list(os.listdir(path))
+    if len(filenames) == 1:
+        data_dicom = os.path.join(path, filenames[0])
+        dataset = dicom.read_file(data_dicom)
+        spect_data = dataset.pixel_array
+    else:
+        raise NotImplementedError('folder contais more than one item')
```

Comment:
If it's always one file, why don't we simply give the filename instead of a folder?